xds/src/main/java/io/grpc/xds/sds/TlsCertificateSecretVolumeSecretProvider.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.sds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ *  Implementation of a file based secret provider.
+ */
+final class TlsCertificateSecretVolumeSecretProvider
+    implements SecretProvider<TlsCertificateStore> {
+
+  private static final Logger logger = Logger
+      .getLogger(TlsCertificateSecretVolumeSecretProvider.class.getName());
+  public static final String PEM = ".pem";
+  public static final String CRT = ".crt";
+
+  private final String path;
+
+  // for now mark it unused
+  @SuppressWarnings("unused")
+  private final String name;
+
+  TlsCertificateSecretVolumeSecretProvider(String path, String name) {
+    this.path = path;
+    this.name = name;
```

Comment:
What about init TlsCertificateStore and ExecutionException here?
```java
    TlsCertificateStore tlsCertificateStore = null;
    ExecutionException executionException = null;
    try {
      tlsCertificateStore = new TlsCertificateStore(
          new FileInputStream(path + PEM),
          new FileInputStream(path + CRT));
    } catch (FileNotFoundException e) {
      executionException = new ExecutionException(e);
    }
    this.tlsCertificateStore = tlsCertificateStore;
    this.executionException = executionException;
```

Reply:
No. The point exactly is to read the contents at the time `get()` is called because the certificate(s) on the file system changes (is dynamic). The `get()` is no longer constrained by the semantics of `Future<>.get()`.

Reply:
i think the method name `get()` has strong connection to `Future`, how about using `getSnapshot()` or something similar name to avoid confusion. `getSnapshot` implicitly tells that the result might be different based on time.